aptos-move/move-examples/token_objects/sources/aptos_token.move

```diff
@@ -0,0 +1,841 @@
+/// This defines a minimally viable token for no-code solutions akin the the original token at
+/// 0x3::token module.
+/// The key features are:
+/// * Base token and collection features
+/// * Creator definable mutability for tokens
+/// * Creator-based freezing of tokens
+/// * Standard object-based transfer and events
+/// * Metadata property type
+///
+/// TODO:
+/// * Fungible of tokens
+module token_objects::aptos_token {
+    use std::error;
+    use std::option::{Self, Option};
+    use std::string::String;
+    use std::signer;
+
+    use aptos_framework::object::{Self, ConstructorRef, Object};
+
+    use token_objects::property_map;
+    use token_objects::token;
+
+    // The token does not exist
+    const ETOKEN_DOES_NOT_EXIST: u64 = 1;
+    /// The provided signer is not the creator
+    const ENOT_CREATOR: u64 = 2;
+    /// Attempted to mutate an immutable field
+    const EFIELD_NOT_MUTABLE: u64 = 3;
+    /// Attempted to burn a non-burnable token
+    const ETOKEN_NOT_BURNABLE: u64 = 4;
+    /// Attempted to mutate a property map that is not mutable
+    const EPROPERTIES_NOT_MUTABLE: u64 = 5;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Storage state for managing the no-code Token.
+    struct AptosToken has key {
+        /// Used to burn.
+        burn_ref: Option<token::BurnRef>,
+        /// Used to control freeze.
+        transfer_ref: Option<object::TransferRef>,
+        /// Used to mutate fields
+        mutator_ref: Option<token::MutatorRef>,
+        /// Used to mutate properties
+        property_mutator_ref: Option<property_map::MutatorRef>,
+        /// Determines if the creator can mutate the description
+        mutable_description: bool,
+        /// Determines if the creator can mutate the name
+        mutable_name: bool,
+        /// Determines if the creator can mutate the uri
+        mutable_uri: bool,
+        /// Determines if the creator can freeze this NFT
+        freezable_by_creator: bool,
+    }
+
+    public entry fun mint(
+        creator: &signer,
+        collection: String,
+        description: String,
+        name: String,
+        uri: String,
+        mutable_description: bool,
```

Comment:
This is a very long list of params. Can we reduce this by:
1. Defaulting all mutability configs to true? Most collection likely go with all trues anyway. We can add functions to allow setting mutability configs to false.
2. property_keys, property_types, property_values: vector<vector<u8>> seem like duplication of PropertyMap, which despite being a bit overly complex, does have proper SDK support that makes it easier to use.

Reply:
https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/plugins/token_client.ts#L92

I'm not sure I entirely follow

also the SDK is borked because why are passing in `property_values` as string if we intend them to be typed and bcs?

the current PropertyMap has implementation issues, we unfortunately need to adjust.

I don't think we should overly bit manipulate an API that will be hidden behind an SDK.

With that being said, I think an update to this is to pull in AptosCollection that pushes configuration to be shared across all tokens.

Reply:
I'm going to write the python sdk this weekend, I think.

Reply:
@lightmark what's the context here about `vector<bool>`